x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/LoggingUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ql.util;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.logging.Level;
+import org.elasticsearch.logging.Logger;
+
+public final class LoggingUtils {
+
+    private LoggingUtils() {}
+
+    /**
+     * Log the failure of a QL query, that failed with an exception.
+     * @param logger The logger to send the failure message to.
+     * @param queryName The type of QL query. Example `EQL query`. It is prepended to the failure message.
+     * @param throwable The exception the query failed with.
+     * @param preamble An optional message to log along the failure. Can be `null`.
+     */
+    public static void logQueryFailure(Logger logger, String queryName, Throwable throwable, @Nullable String preamble) {
+        StringBuilder builder = new StringBuilder(queryName);
```

Comment:
Check the logger value before going through all the effort of assembling the string, which can be costly.
```
Level level = ExceptionsHelper.status(throwable).getStatus() >= 500 ? Level.WARN : Level.DEBUG;
if (logger.isLevel(level)) {
   // assemble string
  logger.log(level, message)
```

Additionally you can use the supplier as long as the exception is accepted:
`logger.log(level, () -> LoggingUtils.createLoggingMessage(error, queryName, throwable, preamble), throwable)`

The advantage of the latter is separating the error message creation from logging.

Reply:
Ah, thanks. Should have checked on available methods before. the `logger.log(..., Throwable)` does pretty much all what's needed, so I've removed the custom logging and wrapper.